test/e2e/airgapped/airgapped_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package airgapped
+
+import (
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-cli/test/e2e/framework"
+	pluginlifecyclee2e "github.com/vmware-tanzu/tanzu-cli/test/e2e/plugin_lifecycle"
+)
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Airgapped-Plugin-DownloadBundle-UploadBundle-Lifecycle]", func() {
+
+	Context("Download plugin bundle, Upload plugin bundle and plugin lifecycle tests with plugin group 'vmware-tkg/v0.0.1'", func() {
+		// Test case: download plugin bundle for plugin-group vmware-tkg/v0.0.1
+		It("download plugin bundle with specific plugin-group vmware-tkg/v0.0.1", func() {
+			err := tf.PluginCmd.DownloadPluginBundle(e2eTestLocalCentralRepoImage, []string{"vmware-tkg/v0.0.1"}, filepath.Join(tempDir, "plugin_bundle_vmware-tkg-v0.0.1.tar.gz"))
+			Expect(err).To(BeNil(), "should not get any error while downloading plugin bundle with specific group")
+		})
+
+		// Test case: upload plugin bundle downloaded using vmware-tkg/v0.0.1 plugin-group to the airgapped repository
+		It("upload plugin bundle that was downloaded using vmware-tkg/v0.0.1 plugin-group to the airgapped repository", func() {
```

Comment:
is upload plugin bundle functionality overrides the repository if any available already or just appends?
if it appends, then better validate repository is empty before uploading.

Reply:
It appends the plugins. Good point, I will add a validation that it was empty before uploading.

Reply:
Actually, for our test cases, we start a new registry every time at `localhost:6001` and that registry doesn't have a plugin inventory database image created until we run first `UploadPluginBundle`. So, any attempt to get the list of the plugins might fail.

So to verify this, I added a check in `BeforeSuite` by verifying air-gapped plugin inventory image doesn't exists before the start of the tests.
```
// Make sure airgapped plugin inventory image doesn't exists before starting the tests
_, _, err = carvelhelpers.GetImageDigest(e2eAirgappedCentralRepoImage)
Expect(err).NotTo(BeNil(), "looks like airgapped plugin inventory image already exists before starting the airgapped tests. This should not be the case as it leads to incorrect test results")
```

Reply:
However, since we use the same `e2eAirgappedCentralRepo` for all use cases, plugins may already be available when the second use case (`'plugin group 'vmware-tmc/v9.9.9'`) is running.
Before each use case, the register might need to be cleaned up or deleted. And tests are supposed to fail, except for the first use case, right? any thoughts?

Reply:
No, these tests are supposed to test the merging of the plugins that we are uploading. If we delete of clean the existing plugins we are not testing the correct behavior we want to test.

The test cases are written in a way that it builds the plugin-inventory by adding different plugin-groups in each phase and eventually builds the entire plugin-inventory in the last phase.

Reply:
Okay got it, thank you.